plugins/ltac/g_auto.mlg

```diff
@@ -150,6 +158,13 @@ TACTIC EXTEND dfs_eauto
     { Eauto.gen_eauto (Eauto.make_dimension p None) (eval_uconstrs ist lems) db }
 END
 
+TACTIC EXTEND bfs_eauto
+| [ "bfs" "eauto" int_or_var_opt(p) auto_using(lems)
+      hintbases(db) ] ->
+    (* todo: need to get the right parameters here *)
+    { Eauto.gen_eauto (Eauto.make_dimension p p) (eval_uconstrs ist lems) db }
```

Comment:
The best way to be correct is to not use this `make_dimension` function but rather to inline its result instead. The result is a tuple of a boolean and an int, the boolean is `true` for dfs and `false` for bfs. The int is the search-depth, but given the syntax that is being deprecated, there was no default search-depth in the bfs case (it always had to be provided). Thus, the simplest solution for now is to not make the search-depth optional in the new syntax. Another solution would be to export the `make_depth` function from the `Eauto` module.

See:
https://github.com/coq/coq/blob/010dfd516839b901ce8e69dffc0c3751564a8ad6/tactics/eauto.ml#L380-L386